components/raftstore/src/store/fsm/store.rs

```diff
@@ -2358,6 +2363,40 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
         let _ = scheduler.schedule(RaftlogGcTask::Purge);
         self.register_raft_engine_purge_tick();
     }
+
+    fn on_check_leader(&self, leaders: Vec<LeaderInfo>, cb: Box<dyn FnOnce(Vec<u64>) + Send>) {
+        let meta = self.ctx.store_meta.lock().unwrap();
+        let regions = leaders
+            .into_iter()
+            .map(|leader_info| {
+                if let Some((term, leader)) = meta.leaders.get(&leader_info.region_id) {
+                    if let Some(region) = meta.regions.get(&leader_info.region_id) {
+                        if *term == leader_info.term
+                            && leader.id == leader_info.peer_id
+                            && util::compare_region_epoch(
+                                leader_info.get_region_epoch(),
+                                region,
+                                true,
+                                false,
+                                false,
+                            )
+                            .is_ok()
```

Comment:
Maybe we should also check the `epoch.ver` here. If the leader split slowly ( perhaps its apply thread is stuck), and the new split region has elected a new leader and writes more data, the old leader can forward the resolved ts while it does not have the most up-to-date data on the region key range.

Reply:
Good catch! Remind me to check the epoch either when using leader lease.